examples/TicTacToe/test/ticTacToeTest.js

```diff
@@ -0,0 +1,179 @@
+      TicTacToeContract = await deployContract(
+        'TicTacToe',
+        owner,
+        gas,
+        deployObj
+      )
+      assert(TicTacToeContract.hasOwnProperty('address'))
+      assert(TicTacToeContract.hasOwnProperty('owner'))
+    })
+  })
+
+  describe('Interact with the contract', () => {
+    before(() => {
+      addressTicTacToe = decodeContractAddress(TicTacToeContract)
+    })
+
+    it('should set player 1 ', async () => {
+      const args = {
+        args: `11`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        TicTacToeContract,
+        'player_1',
+        args,
+        'string'
+      )
+      assert.equal(result, "Game continues. The other player's turn.")
+    })
+
+    it('should set player 2 ', async () => {
+      const args = {
+        args: `12`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        TicTacToeContract,
+        'player_2',
+        args,
+        'string'
+      )
+      assert.equal(result, "Game continues. The other player's turn.")
+    })
+    it('should set player 1 ', async () => {
+      const args = {
+        args: `33`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        TicTacToeContract,
+        'player_1',
+        args,
+        'string'
+      )
+      assert.equal(result, "Game continues. The other player's turn.")
+    })
+
+    it('should set player 2 ', async () => {
+      const args = {
+        args: `23`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        TicTacToeContract,
+        'player_2',
+        args,
+        'string'
+      )
+      assert.equal(result, "Game continues. The other player's turn.")
+    })
+
+    it('should set player 1 ', async () => {
+      const args = {
+        args: `31`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        TicTacToeContract,
+        'player_1',
+        args,
+        'string'
+      )
+      assert.equal(result, "Game continues. The other player's turn.")
+    })
+
+    it('should set player 2 ', async () => {
+      const args = {
+        args: `21`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        TicTacToeContract,
+        'player_2',
+        args,
+        'string'
+      )
+      assert.equal(result, "Game continues. The other player's turn.")
+    })
+
+    it('should win player 1 ', async () => {
+      const args = {
+        args: `32`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        TicTacToeContract,
+        'player_1',
+        args,
+        'string'
+      )
+      assert.equal(result, 'You are the winer! Congratulations player 1')
+    })
+  })
+})
```

Comment:
you don't have test for player 2 is a winner and no one is a winner.
if in function player_2 by some reason you change the player /check_win(position, 1)/... check_win should check moves on player 1 and will return that winner is a player 1...